yt_dlp/extractor/podbayfm.py

```diff
@@ -1,82 +1,76 @@
+        'title': props['title'],
+        'url': props['mediaURL'],
+        'ext': 'mp3',
+        'thumbnail': image_props.get('url'),
+        'timestamp': props.get('timestamp'),
+        'duration': try_get(props, lambda p: int(p['duration'], 10))
+    }
 
 
 class PodbayFMIE(InfoExtractor):
     _VALID_URL = r'https?://podbay\.fm/p/[^/]*/e/(?P<id>[^/]*)/?(?:[\?#].*)?$'
     _TESTS = [{
         'url': 'https://podbay.fm/p/behind-the-bastards/e/1647338400',
-        'md5': 'be0fd3275a12e9acf6d89037b9dbea04',
+        'md5': '98b41285dcf7989d105a4ed0404054cf',
         'info_dict': {
             'id': '1647338400',
-            'ext': 'mp3',
             'title': 'Part One: Kissinger',
+            'ext': 'mp3',
+            'thumbnail': r're:^https?://.*\.jpg',
+            'timestamp': 1647338400,
             'duration': 5001,
-        },
-        'params': {
-            'skip_download': True,
+            'upload_date': '20220315',
         },
     }]
 
-    def _parse_jwt_add_padding(self, token, video_id):
-        b64 = token.split('.')[1]
-        b64 += (4 - (len(b64) % 4)) * '='
-        json = urlsafe_b64decode(b64).decode('utf-8')
-        return self._parse_json(json, video_id)
-
-    def _result_from_props(self, props, episode_id):
-        from pprint import pprint
-        pprint(props)
-        image_props = self._parse_jwt_add_padding(props['image'], episode_id)
-        return {
-            'id': props.get('podcast_id') or episode_id,
-            'title': props['title'],
-            'url': props['mediaURL'],
-            'ext': 'mp3',
-            'thumbnail': image_props['url'],
-            'timestamp': props['timestamp'],
-            'duration': int(props['duration'], 10),
-        }
-
     def _real_extract(self, url):
         episode_id = self._match_id(url)
         webpage = self._download_webpage(url, episode_id)
-        data = self._parse_json(
-            self._html_search_regex(r'<script .*?\bid="__NEXT_DATA__".*?>(.*)</script>', webpage, 'JSON data'),
-            episode_id)
-        props = data['props']['pageProps']['episode']
-        return self._result_from_props(props, episode_id)
+        data = self._search_nextjs_data(webpage, episode_id)
+        return result_from_props(data['props']['pageProps']['episode'], episode_id)
 
 
-class PodbayFMChannelIE(PodbayFMIE):
+class PodbayFMChannelIE(InfoExtractor):
     _VALID_URL = r'https?://podbay\.fm/p/(?P<id>[^/]*)/?(?:[\?#].*)?$'
     _TESTS = [{
         'url': 'https://podbay.fm/p/behind-the-bastards',
+        'info_dict': {
+            'id': 'behind-the-bastards',
+            'title': 'Behind the Bastards',
+        },
     }]
+    _PAGE_SIZE = 10
+
+    def _fetch_page(self, channel_id, pagenum):
+        page = self._download_json(
+            'https://podbay.fm/api/podcast?reverse=true&page=' + str(pagenum) + '&slug=' + channel_id,
+            channel_id)['podcast']
+        return [result_from_props(e, None) for e in page['episodes']]
```

Comment:
Normally, you'd return a url_result. Here you are returing the direct video result, and I can see that makes sense. But when you do this, you need to explicitly set the correct `webpage_url`, `extractor_key` and `extractor`